src/connections.jl

```diff
@@ -180,10 +180,21 @@ function _connect_poll(conn::Ptr{libpq_c.PGconn}, timer::Timer, timeout::Real)
         # PostgreSQL docs say we can't assume that the socket will remain the same while
         # connecting
         try
-            if p_state == libpq_c.PGRES_POLLING_WRITING
-                wait(socket(conn); writable=true)
-            elseif p_state == libpq_c.PGRES_POLLING_READING
-                wait(socket(conn); readable=true)
+            wait_write = p_state == libpq_c.PGRES_POLLING_WRITING
+            wait_read = p_state == libpq_c.PGRES_POLLING_READING
+
+            # poll_fd disables timeout at <0, so if we have 0 make sure it's actually less
+            if wait_write || wait_read
+                event = poll_fd(
+                    socket(conn),
+                    timeout - 0.001;
```

Comment:
Shouldn't this be done outside the loop as:
```julia
if timeout == 0
   timeout = -1
```
?

Reply:
Yeahhhh